Framework/CurveFitting/src/IMWDomainCreator.cpp

```diff
@@ -303,13 +303,16 @@ boost::shared_ptr<API::Workspace> IMWDomainCreator::createOutputWorkspace(
   textAxis->setLabel(0, "Data");
   textAxis->setLabel(1, "Calc");
   textAxis->setLabel(2, "Diff");
-
+  std::unordered_map<std::string, int> functionCount;
   // Add each calculated function
   auto iend = functionsToDisplay.end();
   size_t wsIndex(1); // Zero reserved for data
   for (auto it = functionsToDisplay.begin(); it != iend; ++it) {
-    if (wsIndex > 2)
-      textAxis->setLabel(wsIndex, (*it)->name());
+    if (wsIndex > 2) {
+      std::string name = (*it)->name();
+      ++functionCount[name];
+      textAxis->setLabel(wsIndex, name + std::to_string(functionCount[name]));
```

Comment:
Do unique names have to have the index '1'?

Reply:
sorry I do not understand the question. Are you asking:

1. Why it is in the if statement with `wsIndex > 2`? - if it is 1 then there is no need for unique names
2. If it will always append `1` to the name? - line 312 will mean we have a count for the number of occurrences

Reply:
If you have a single LinearBackground it may make sense not to give it any index at all.

Reply:
ah ok, yes it would but if we also had two GaussOsc then it would be inconsistent naming. It would also require us to determine the number of instances of each function prior to writing them. which would add complexity to the code.

Reply:
It is not much of a complexity, just an extra loop. And it is just my opinion.

Reply:
I'd have to side with Roman on this one. The code is simpler, yes, but the users will be puzzled to see a "1" suffix when they're not expecting one. I'd rather go with a simpler user-interface at the expense of increasing the complexity of the source code.